scripts/genpinctrl/genpinctrl.py

```diff
@@ -507,13 +504,24 @@ def main(cube_path, output):
                         if af["name"] not in entries:
                             entries[af["name"]] = list()
 
+                        # Define the signal mode using, by priority order:
+                        # 1- the config mode (ie: "af["mode"]")
+                        # 2- the inferred mode (an alternate function was found)
+                        if af.get("mode") or family == "STM32F1":
+                            signal["mode"] = af["mode"]
+                        else:
```

Comment:
Actually, can this be compressed into a single if/elif/else block?

Reply:
Sure, but wanted to make it clear.